test/units/module_utils/parsing/test_convert_bool.py

```diff
@@ -10,49 +10,53 @@
 
 
 class TestBoolean:
-    def test_bools(self):
-        assert boolean(True) is True
-        assert boolean(False) is False
-
-    def test_none(self):
-        with pytest.raises(TypeError):
-            assert boolean(None, strict=True) is False
-        assert boolean(None, strict=False) is False
-
-    def test_numbers(self):
-        assert boolean(1) is True
-        assert boolean(0) is False
-        assert boolean(0.0) is False
-
-# Current boolean() doesn't consider these to be true values
-#    def test_other_numbers(self):
-#        assert boolean(2) is True
-#        assert boolean(-1) is True
-#        assert boolean(0.1) is True
-
-    def test_strings(self):
-        assert boolean("true") is True
-        assert boolean("TRUE") is True
-        assert boolean("t") is True
-        assert boolean("yes") is True
-        assert boolean("y") is True
-        assert boolean("on") is True
-
-    def test_junk_values_nonstrict(self):
-        assert boolean("flibbity", strict=False) is False
-        assert boolean(42, strict=False) is False
-        assert boolean(42.0, strict=False) is False
-        assert boolean(object(), strict=False) is False
-
-    def test_junk_values_strict(self):
-        with pytest.raises(TypeError):
-            assert boolean("flibbity", strict=True) is False
-
-        with pytest.raises(TypeError):
-            assert boolean(42, strict=True) is False
-
-        with pytest.raises(TypeError):
-            assert boolean(42.0, strict=True) is False
-
-        with pytest.raises(TypeError):
-            assert boolean(object(), strict=True) is False
+    @pytest.mark.parametrize("test,expected", [
+        (True, True),
+        (False, False),
+        (1, True),
+        (0, False),
+        (0.0, False),
+    ])
+    def test_numbers(self, test, expected):
+        assert boolean(test) is expected
+
+    @pytest.mark.skip(reason="Current boolean() doesn't consider these to be true values")
+    @pytest.mark.parametrize("test", [
+        (2),
+        (-1),
+        (0.1),
+    ])
```

Comment:
The parentheses here and in the other tests are redundant:
```suggestion
    @pytest.mark.parametrize("test", (2, -1, 0.1))
```